04/2_print_string_buffer/buffered_print.c

```diff
@@ -0,0 +1,42 @@
+#include <stdio.h>
+#include <string.h>
+
+#include "buffered_print.h"
+
+#define BUFFER_LENGTH (32)
+
+static size_t s_buffer_index = 0u;
+static char s_buffer[BUFFER_LENGTH];
+
+void buffered_print(const char* src)
+{
+    size_t num_left;
+    const char* P = src;
+
+    num_left = strlen(src);
+
+    while (num_left > 0) {
+        const size_t NUM_CHARS_TO_BUFFER = 

```

Comment:
don't do capital here.

do 

1 assign
2 if compare to update

i don't wanna see this calc twice

`BUFFER_LENGTH - 1 - s_buffer_index`